pkg/services/grpcserver/service.go

```diff
@@ -44,9 +44,9 @@ type gPRCServerService struct {
 
 func ProvideService(cfg *setting.Cfg, features featuremgmt.FeatureToggles, authenticator interceptors.Authenticator, tracer tracing.Tracer, registerer prometheus.Registerer) (Provider, error) {
 	s := &gPRCServerService{
-		cfg:         cfg,
+		cfg:         cfg.GRPCServer,
 		logger:      log.New("grpc-server"),
-		enabled:     features.IsEnabledGlobally(featuremgmt.FlagGrpcServer),
+		enabled:     features.IsEnabledGlobally(featuremgmt.FlagGrpcServer) || cfg.GRPCServer.Enabled,
```

Comment:
Agreed. Since removing the config option here has no effect (I call Run directly), I'll add a comment about it for when the feature flag is removed.